packages/scratch-vm/scripts/codeGeneration/index.ts

```diff
@@ -1,41 +1,101 @@
-import { readFileSync } from "fs";
+import { readFileSync, writeFileSync, mkdirSync, existsSync, copyFileSync } from "fs";
 import path = require("path");
 import { ExtensionMenuDisplayDetails } from "../../src/typescript-support/types";
 
 const relativePathToExtensionDir = ["..", "..", "src", "extensions"];
-const baseExtensionsDir = path.resolve(__dirname, ...relativePathToExtensionDir);
-const getPathToExtension = (extensionId: string) => path.join(baseExtensionsDir, extensionId);
 
-const relativePathToIndexFile = ["..", "..", "..", "scratch-gui", "src", "lib", "libraries", "extensions", "index.jsx"]; 
+const relativePathToIndexFile = ["..", "..", "..", "scratch-gui", "src", "lib", "libraries", "extensions", "index.jsx"];
+const relativePathToAssetsFolder = ["..", "..", "..", "scratch-gui", "src", "extension-gallery-assets"];
 const guiIndexFile = path.resolve(__dirname, ...relativePathToIndexFile);
 
-const guardFlagStart = "/* CODE GEN GUARD START: ";
-const guardFlagEnd = "/* CODE GEN GUARD END: ";
+/**
+ * has to match the code generation guards in the file specified by
+ * the path in {@link relativePathToIndexFile}
+ */
+const guardFlagStart = "/* CODE GEN GUARD START:";
+const guardFlagEnd = "/* CODE GEN GUARD END:";
+const codeGuardStringEnd = '-- Please do not edit code within guards */';
 
-const getGuards = (identifier: string): [start: string, end: string] => [`${guardFlagStart} ${identifier}`, `${guardFlagEnd} ${identifier}`, ]
```

Comment:
You didn't like my tuple labels?? 

Also just FYI, I think it's totally good that you added the `codeGuardStringEnd` to these, but the reason I left them off in the first place was because I'm reasonably confident the `/* CODE GEN GUARD START:` text will never change, but It's possible we might want to revise the `-- Please do not edit code within guards ` message (say, if people aren't understanding or respecting it). And of course, if we left it off, we would've checked that the line just started with the guard text instead of matched it exactly. 

But no need to change it as I don't think the code gen guards will change. Just wanted to mention it as food for thought when making a code gen system / this kind of framework, where you give yourself some flexibility to revise things later and don't need to remember every last place you added text.

Reply:
I think I remember seeing an error with the names but I didn't mind them. Also, I already am searching for the code guards by finding a substring within the line so we can discuss changing the inclusion of the `codeGuardStringEnd`

Reply:
lol I was just joshing about the tuple labels. Better to have them, but no reason to fight to make them work.

And sweet -- well then I would recommend dropping them, for maintainability going forward

Reply:
restored the naming on the tuple, and addressed the second part of the comment (don't include the `Please don't edit...` phrase in the substring search).